CHANGELOG.next.asciidoc

```diff
@@ -44,6 +44,10 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Update Salesforce module to use new Salesforce input. {pull}37509[37509]
 - Tag events that come from a filestream in "take over" mode. {pull}39828[39828]
 - Fix high IO and handling of a corrupted registry log file. {pull}35893[35893]
+- Journald: removed configuration options `include_matches.or`, `include_matches.and`, `backoff`, `max_backoff`, `cursor_seek_fallback`. {pull}40061[40061]
```

Comment:
What is the replacement functionality for people using `include_matches.or` or `include_matches.and`?

What happens for configurations that specify these parameters today? Are the parameters ignored?

Reply:
Reading the docs changes it looks like `include_matches.match` which seems simpler

Reply:
> What is the replacement functionality for people using include_matches.or or include_matches.and?

There isn't a perfect, one-to-one replacement, but the current implementation has its own problems as I described on https://github.com/elastic/beats/issues/37086#issuecomment-2195237151.

TL;DR: `include_matches.or` or `include_matches.and` are hard to understand and cannot be used to make any arbitrary boolean logic. What is supported is not clearly documented. This is a breaking change and specifying either `include_matches.or` or `include_matches.and` will cause the config validation to fail.

Long version.

The implementation on this PR, simply gets the elements from `include_matches.match` and passes it directly to journalctl as its `MATCHES`. Currently makes keys provided more than once to be interpreted as a logical OR and different keys as logical AND.

`journalctl` allows for a bit more complex expressions, but it is still not a fully arbitrary boolean logic.

I added some test for the matchers that show the current possibilities for boolean "expressions": https://github.com/elastic/beats/blob/68843feeeaaff1f36ecd2313dfa12692a849d03d/filebeat/input/journald/input_test.go#L211-L268, there is also some examples in the documentation.